webapp/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.data.GiftCertificateDto;
+import com.epam.esm.entities.GiftCertificate;
+import com.epam.esm.service.CertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/certificates")
```

Comment:
Please, read more carefully article, which i sent you in previous comment. Pay attention on versioning

Reply:
Should I place version in controller or in DipatcherServlet config?

Reply:
you can decide for yourself